### DISCUSSION TITLE
Any library template/builder?

### DISCUSSION THREAD

**Discussion by jacobp100** [reactwg/react-native-new-architecture#20] [Libraries]
I've used react-native-create-library in the past - are there any boilerplate projects or builders that support fabric?

**Comment by leotm** [comment]
I'm working on a bleeding-edge [react-native-template-new-architecture](https://github.com/leotm/react-native-template-new-architecture) atm which comes with the new arch enabled you mention ^

It's a bit more than boilerplate, but you can always downgrade deps to stable, linting from _all_ to _recommended_, etc

Gradually building a CLI to make this easier, any suggestions welcome

And looking into codegen'ing e.g. [a native Bootsplash](https://github.com/leotm/react-native-template-typescript/pull/379), since imo it's better to keep the boilerplate minimal

I'll go ahead and post [here](https://github.com/reactwg/react-native-new-architecture/discussions/21)

**Comment by cortinico** [comment]
> are there any boilerplate projects or builders that support fabric?

We do have an official sample app that uses the New Architecture across different versions of React Native: https://github.com/react-native-community/RNNewArchitectureApp

For libraries, we're working on similar material and documentation that we hope to release soon.

As for templates, I'm not aware of any solution that works with the new arch out of the box. I'm happy to look into solutions/suggestions if you have any to share.

**Comment by kelset** [comment]
There's also this library from @alon7 : https://github.com/talknagish/react-native-turbo-starter

**Comment by satya164** [comment]
[`create-react-native-library`](https://github.com/callstack/react-native-builder-bob) now supports generating a project for libraries using turbo modules (both with and without backward compatibility). It's still experimental so I'd love any feedback.
